db/getItemsOnList.go

```diff
@@ -0,0 +1,39 @@
+package db
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/mount-joy/thelist-lambda/data"
+)
+
+func (d *db) GetItemsOnList(listID *string) (*[]data.Item, error) {
+	input := &dynamodb.QueryInput{
+		ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
+			":id": {S: listID},
+		},
+		KeyConditionExpression: aws.String("ListId = :id"),
+		TableName:              aws.String(dbTableNameItems),
+	}
+
+	result, err := d.session.Query(input)
+	if err != nil {
+		return nil, err
+	}
+	if result.Items == nil {
+		return nil, nil
```

Comment:
For a `Get` where the got thing is `nil`, should we not be returning a not found error?